usermanagement/src/main/kotlin/com/story/usermanagement/storage/UserDynamoDto.kt

```diff
@@ -0,0 +1,32 @@
+package com.story.usermanagement.storage
+
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPartitionKey
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbSecondaryPartitionKey
+
+const val emailAddressIndex = "email-address-index"
+
+@DynamoDbBean
+@DynamoTable(baseTableName = "users")
+@Provisions(readCapacityUnits = 10L, writeCapacityUnits = 10L)
+class UserDynamoDto {
+
+    @get:DynamoDbPartitionKey var id: Long = 0
```

Comment:
If you are going to use single table design please read  pros and cons then design your schema properly. https://aws.amazon.com/blogs/database/single-table-vs-multi-table-design-in-amazon-dynamodb/

Reply:
Also you might want to use https://github.com/michaellavelle/spring-data-dynamodb to ease the development.